runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -1184,4 +1184,140 @@ function LanguageTree:language_for_range(range)
   return self
 end
 
+--- Get highlight information from parser.
+---
+---@return {range:Range4, hl_group:string, priority:integer, conceal:string?, spell:boolean?, url:string?}[]
+function LanguageTree:get_highlight()
+  local highlights = {}
+  self:for_each_tree(function(tree, ltree)
+    local source = ltree:source()
+    local hl_query = query.get(ltree:lang(), 'highlights')
+
+    if not hl_query then
+      return
+    end
+
+    local highlighter = require('vim.treesitter.highlighter')
+    for info in highlighter._iter_highlight_info(hl_query, tree:root(), source, 0, -1) do
+      info.range = { Range.unpack4(info.range) }
+      table.insert(highlights, info)
+    end
+  end)
+  return highlights
+end
+
+---@nodoc
+---@class vim.treesitter.LanguageTree.highlight_resolve_priority.obj
+---@field group string
+---@field range integer[]
+---@field priority integer
+---@field type 'start'|'end'
+---@field start? table
+---@field subpriority integer
+---@field info table
+
+---@param stack vim.treesitter.LanguageTree.highlight_resolve_priority.obj[]
+---@param subpriority number
+---@return number
```

Comment:
Shouldn't this be(?):
```suggestion
---@return integer
```